sbi/neural_nets/density_estimators/flow.py

```diff
@@ -52,12 +65,60 @@ def sample(self, sample_shape: Shape, condition: Tensor) -> Tensor:
         Returns:
             Samples.
         """
+        self._check_for_invalid_condition_shape(condition)
+
+        num_samples = torch.Size(sample_shape).numel()
+        x_dim = len(self._x_shape)
+
+        if len(condition.shape) == x_dim:
+            # nflows.sample() expects conditions to be batched.
+            condition = condition.unsqueeze(0)
+            samples = self.net.sample(num_samples, context=condition).reshape(
+                (*sample_shape, -1)
+            )
+        else:
+            # For batched conditions, we need to reshape the conditions and the samples
+            batch_dims = condition.shape[:-x_dim]
```

Comment:
we should be consistent with the variable names: `x_dim` vs `x_dims` vs `batch_dims` vs `batch_dim`. I thing the plural makes it clearer it's just an integer count of the number of dimensions.

Reply:
I see will adapt plural and stick to `condition_dims` and `batch_dims`